src/compute/src/render/reduce.rs

```diff
@@ -720,8 +726,9 @@ where
     // We arrange the inputs ourself to force it into a leaner structure because we know we
     // won't care about values.
     let partial = collection
-        // TODO(#16549): Use explicit arrangement
-        .consolidate()
+        .map(|k| (k, ()))
+        .arrange_named::<RowKeySpine<_, _, _>>("Arranged ReduceMonotonic input")
+        .as_collection(|d: &(Row, Vec<Row>), _| d.clone())
```

Comment:
Total drive-by, but: I noticed that some of these operator chains now become harder to read. For example, here it is not immediately obvious that we are simply performing a consolidation anymore. Is there space for expressive wrappers in `CollectionExt`? In this case, I think we could provide a `consolidate1` (with a better name!) that's generic over the spine type and then write something like `collection.consolidate1::<RowKeySpine<_, _, _>>()` here.

Reply:
This is exactly the kind of feedback I was hoping for, since it's not clear to me what "reads" well to the experienced eye in this code.  I did ask Moritz before about possibly extending DD with the generic type or adding our own helper, and there was some obstacle, but it sounds like Frank is going to add this to DD, so perhaps that's that, and I can hold off on merging this until that becomes available.